eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BitcoinCoreClient.scala

```diff
@@ -191,9 +193,10 @@ class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient) extends OnChainWall
       // with a verbosity of 0, getblock returns the raw serialized block
       block <- rpcClient.invoke("getblock", blockhash, 0).collect { case JString(b) => Block.read(b) }
       prevblockhash = block.header.hashPreviousBlock.reversed()
-      res <- block.tx.asScala.find(tx => tx.txIn.asScala.exists(i => i.outPoint.txid == txid && i.outPoint.index == outputIndex)) match {
-        case None => lookForSpendingTx(Some(prevblockhash), txid, outputIndex)
+      res <- block.tx.asScala.find(tx => tx.txIn.asScala.exists(i => i.outPoint.txid == KotlinUtils.scala2kmp(txid) && i.outPoint.index == outputIndex)) match {
```

Comment:
I wonder what the cost of implicit conversions to scala collections is here (potential issue is the creation of many short-lived objects), it would be more efficient to use java iterators and do the txid conversion only once ?

Reply:
It's definitely worth investigating, but we shouldn't make the code less readable and less idiomatic without benchmarking this first. To be honest I don't think this should be in this PR, if we want to improve our allocation strategy by reducing kotlin/scala conversion objects, it should be part of a PR where we apply that to many places in the codebase, right?